app/pages/admin.js

```diff
@@ -46,16 +70,39 @@ export default function Admin() {
                 </div>
                 <div className="flex w-screen mx-4 mt-2">
                   <div>
-                    {links.map((link, index) => (
-                      <div key={index}>
-                        {amountOfLinks > 0 &&  (
-                        <div>
-                          <span>Name</span>
-                          <input name="name" type="text" id="name" required  onBlur={change(index)}/>
-                          <span>Link</span>
-                          <input name="link" type="text" id="link" required />
-                        </div>
-                          )}
+                    {[...links.keys()].map((key) => (
+                      <div key={key} id = {key}>
+                        {links.size > 0 && (
```

Comment:
This check you probably wanna do before iteration like so, also do take in mind that .entries and .map rely on the array having length to iterate over... so you don't need to precheck for length to avoid rendering.
```javascript
{links.size > 0 && links.entries().map(...)
```